jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/ImageName.java

```diff
@@ -273,7 +273,7 @@ private void doValidate() {
         List<String> errors = new ArrayList<>();
         // Strip off user from repository name
         String user = inferUser();
-        String image = user != null ? repository.substring(user.length() + 1) : repository;
+        String image = extractImageName(repository);
```

Comment:
Instead of calling this method in `doValidate`, you should call it in constructor where `repository` is getting initialized.

At the moment, `repository` also contains trailing `:` from empty tag. Try running adding this test and running it: 

```java
    @Test
    void withTagAndDigest() {
        ImageName imageName = new ImageName("registry.example.com/hello-world:@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9");

        assertThat(imageName.getTag()).isNull();
        assertThat(imageName.getDigest()).isEqualTo("sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9");
        assertThat(imageName.getNameWithoutTag()).isEqualTo("registry.example.com/hello-world");
    }
```